### PR TITLE
[plugin.video.ampya] 0.1.10

### PR DESCRIPTION
**Pull request by tyl0re**
### Description
New Plugin for Playing Music Videos from the GermanMusicVideo Platform http://ampya.com/

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.ampya/addon.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    </requires>
+    <extension point="xbmc.python.pluginsource" library="default.py">
+        <provides>video</provides>
+    </extension>
+  <extension point="xbmc.addon.metadata">
+    <summary lang="en">Musicvideos from the German Webpage http://ampya.com/</summary>
+    <summary lang="de">Musicvideos der Seite http://ampya.com/</summary>
+    <description lang="en">Musicvideos from the German Webpage http://ampya.com/</description>
+    <description lang="de">Musicvideos der Seite http://ampya.com/</description>
+    <website></website>
```
Comment (phil65) [inline plugin.video.ampya/addon.xml]: license tag.

File: plugin.video.ampya/default.py [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+  ok = True
+  liz = xbmcgui.ListItem(name, iconImage=icon, thumbnailImage=iconimage)
+  liz.setInfo(type="Video", infoLabels={"Title": name, "Plot": desc})
+  if useThumbAsFanart:
+    if not iconimage or iconimage==icon or iconimage==defaultThumb:
+      iconimage = defaultBackground    
+    liz.setArt({ 'fanart': iconimage })
+  else:
+    liz.setArt({ 'fanart': defaultBackground })  
+  xbmcplugin.setContent(int(sys.argv[1]), 'musicvideos')
```
Comment (phil65) [inline plugin.video.ampya/default.py]: hmm setContent() doesnt make any sense here... you dont call that for every item, but just once for any plugin call.

File: plugin.video.ampya/default.py [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+def addLink(name, url, mode, iconimage, duration="", desc="",artist_id="",genre="",shortname="",production_year=0,zeit=0,liedid=0):  
+  cd=addon.getSetting("password")  
+  u = sys.argv[0]+"?url="+urllib.quote_plus(url)+"&mode="+str(mode)
+  ok = True
+  liz = xbmcgui.ListItem(name, iconImage=defaultThumb, thumbnailImage=iconimage)
+  liz.setInfo(type="Video", infoLabels={"Title": name, "Plot": desc, "Genre": genre,"Sorttitle":shortname,"Dateadded":zeit,"year":production_year })
+  liz.setProperty('IsPlayable', 'true')
+  liz.addStreamInfo('video', { 'duration' : duration })
+  liz.setArt({ 'fanart': iconimage }) 
+  xbmcplugin.setContent(int(sys.argv[1]), 'musicvideos')
```
Comment (phil65) [inline plugin.video.ampya/default.py]: same here. if you want to set the mediatype of a listitem, use setInfo()

File: plugin.video.ampya/default.py [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+  commands = []
+  listatrist = "plugin://plugin.video.ampya/?mode=songs_from_artist&url="+urllib.quote_plus(str(artist_id))
```
Comment (phil65) [inline plugin.video.ampya/default.py]: typo (atrist)

File: plugin.video.ampya/default.py [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+  ok = True
+  liz = xbmcgui.ListItem(name, iconImage=defaultThumb, thumbnailImage=iconimage)
+  liz.setInfo(type="Video", infoLabels={"Title": name, "Plot": desc, "Genre": genre,"Sorttitle":shortname,"Dateadded":zeit,"year":production_year })
+  liz.setProperty('IsPlayable', 'true')
+  liz.addStreamInfo('video', { 'duration' : duration })
+  liz.setArt({ 'fanart': iconimage }) 
+  xbmcplugin.setContent(int(sys.argv[1]), 'musicvideos')
+  commands = []
+  listatrist = "plugin://plugin.video.ampya/?mode=songs_from_artist&url="+urllib.quote_plus(str(artist_id))  
+  listsimiliar = "plugin://plugin.video.ampya/?mode=list_similiar&url="+urllib.quote_plus(str(liedid))
```
Comment (phil65) [inline plugin.video.ampya/default.py]: similar

File: plugin.video.ampya/default.py [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+  liz = xbmcgui.ListItem(name, iconImage=defaultThumb, thumbnailImage=iconimage)
+  liz.setInfo(type="Video", infoLabels={"Title": name, "Plot": desc, "Genre": genre,"Sorttitle":shortname,"Dateadded":zeit,"year":production_year })
+  liz.setProperty('IsPlayable', 'true')
+  liz.addStreamInfo('video', { 'duration' : duration })
+  liz.setArt({ 'fanart': iconimage }) 
+  xbmcplugin.setContent(int(sys.argv[1]), 'musicvideos')
+  commands = []
+  listatrist = "plugin://plugin.video.ampya/?mode=songs_from_artist&url="+urllib.quote_plus(str(artist_id))  
+  listsimiliar = "plugin://plugin.video.ampya/?mode=list_similiar&url="+urllib.quote_plus(str(liedid))  
+  commands.append(( "Mehr vom Künstler" , 'ActivateWindow(Videos,'+ listatrist +')'))
```
Comment (phil65) [inline plugin.video.ampya/default.py]: please translate.

**Comment by tyl0re** [comment]
Should be al corrected. Last Pull is 0.1.11 (Copypasted wrong sorry). Last Pull was because i forgogt to delete the empty settings again

**Pull request merged**